transport/http2_server.go

```diff
@@ -234,6 +235,16 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 	s.windowHandler = func(n int) {
 		t.updateWindow(s, uint32(n))
 	}
+	if stats.On() {
+		inHeader := &stats.InHeader{
+			FullMethod: s.method,
+			RemoteAddr: t.conn.RemoteAddr(),
+			LocalAddr:  t.conn.LocalAddr(),
```

Comment:
There is a chance the t.conn is broken when these are called so that they cannot get the values. You should probably cache them when the connection is setup.